### ISSUE TITLE
Please tag current project status before next change

### ISSUE THREAD

**Issue by normai** [eosterberg/terminaljs#15]
Please tag the current version **before** you edit files or merge pull request.

Reason: References from forks to original sources will loose facts, if the sources change.

BTW. Perhaps you like have a look at [my fork](https://github.com/normai/terminaljs), which includes further features. I have omitted to make a pull request, because the changes may be too sweeping, are they?

Thanks in advance for anybody who will take care of this project.

**Comment by normai** [comment]
Hello Erik and listeners!

I made a release of your long term master version: [v2.0 — Final 2017](https://github.com/normai/terminaljs/releases/tag/v2.0). Hopefully, I did not bother you with my request to do so. The issue is finished now.

I find terminal.js a cool project and like to keep it alive. Not sure yet how to proceed. I have the idea to just create a new project Terminal2Js, import the original Git history, and voilà. My current release is [v0.2.7 — Process pending pull requests](https://github.com/normai/terminaljs/releases/tag/v0.2.7).

Hello @commanddotcom, are you interested in working together on a low-traffic long-term project?

Bye, Norbert

**Comment by commanddotcom** [comment]
Hi @normai ! Thanks for messaging.

I think Eric doesn't have time for the project now. He gave me access few weeks ago but I didn't have time to wrap around my head all the forks and pull requests yet. I'll take a look at your suggestions later this week and I'll let you know.

**Comment by normai** [comment]
Hi  @commanddotcom, I have analysed the pull requests and implemented some of them.

What I am doing might not exactly match your plans. We should find this out. It were nice, if we can keep the project as a single one instead two — but not at any price. If our plans diverge, so should the projects.

My coding style shall not be an obstacle. I am aware, that e.g. the funny timestamps cluttering the code may scare off others. On such formal questions, I am of course willing to compromise, e.g. if purge the timestamps.

BTW. Your XHR code is baked in. But tested only in so far, that it does not break the other features, not whether it will work with your backend. Getting feedback about this were most interesting for me. No hurry.

Bye, N.

**Comment by commanddotcom** [comment]
@normai just to keep you heads up:

I revisited code in **eosterberg/terminaljs** and PRs. I made decision to convert this library to NPM package, but version for CDN might be build as well later on. 

I really enjoyed your trick with `beep.mp3` (base64 converting), but base64 code might be considered potential security issue by other developers. Further on: it is not flexible, it makes library significantly bigger and it is hard to find any important use-case for it (it brings some joy but it's all it does). I made decision to drop this functionality. There are a lot of NPM's packages for media that can be used alongside with TerminalJS. 

Right now you can find version 3.0.0 on the master and [2.0 Final Release 2014](https://github.com/eosterberg/terminaljs/releases/tag/2.0).

Although formatting for 3.0.0. is different and I make some refactoring the basic logic is pretty much the same. So, if you have any suggestion about possible improvements please, feel free to create PR, open issues or contact me in private.

**Comment by normai** [comment]
Hello @commanddotcom! Wow, your work looks very good. Congratulations! And thank you for the tags!

Had I knew that you continue that way, I possibly had not pushed my own fork so far. Now the two projects are far off, and pull requests will make little sense. Anyway, I will bring it to some point where it can rest a while.

I am experimenting with the name, to distinguish the two projects, currently I am trying '[Termjnal](https://github.com/normai/terminaljs)'.

> base64 code might be considered potential security issue by other developers

Interesting point. As I found in a hurry, decoding functions e.g. in PHP are an issue. I will keep that on the radar, though I cannot imagine any problem (yet?) in the JavaScript case. The library being single-file and dependency-free is high priority for me.

> .. it makes library significantly bigger ..

Indeed. MIDI were the best solution as about size. I have investigated on that in a session [MidiProbe](http://localhost/trekta/midiprobedev/trunk/midiprobe/). But MIDI will break IE compatibility, which I want maintain still for a while. For this reason I have even prefered the 42 KB MP3 file over the 25 KB OGG file.

As I have seen, you have thrown out IE completely anyway.

I was well on the go, but now want finish the session in one or two days. I am looking forward, to see the outcome of your session. Then we can watch at the matter from bird's eye view, and ponder, how we can support one another.

Keep the good work going.
